jsonstore/jsonstore.go

```diff
@@ -524,3 +583,73 @@ func (app *JSONStoreApplication) Commit() types.ResponseCommit {
 func (app *JSONStoreApplication) Query(reqQuery types.RequestQuery) (resQuery types.ResponseQuery) {
 	return
 }
+
+func (app *JSONStoreApplication) InitChain(req types.RequestInitChain) types.ResponseInitChain {
+	fmt.Println("calling InitChain")
+	validators := req.GetValidators()
+	var mintValidators []interface{}
+	if validators != nil {
+		for _, element := range validators {
+			validator := Validator{
+				ID:        bson.NewObjectId(),
+				Name:      "validatorName",
+				PublicKey: element.GetPubKey(),
+				Votes:     []bson.ObjectId{},
+			}
+			mintValidators = append(mintValidators, validator)
+		}
+		dbErr := db.C("validators").Insert(mintValidators...)
+		if dbErr != nil {
+			panic(dbErr)
+		}
+	}
+	return types.ResponseInitChain{}
+}
+
+func (app *JSONStoreApplication) EndBlock(req types.RequestEndBlock) types.ResponseEndBlock {
+	// TODO below solution needs confirmation through testing on multiple nodes.
+	// according to documentation "To add a new validator or update an existing one,
+	// simply include them in the list returned in the EndBlock response.
+	// To remove one, include it in the list with a power equal to 0."
+	// That means below solution may not be removing any validators atm!
+
+	// TODO I've noticed that the order of elements with the same vote value is not nondeterministic
+	// which means last active can be swapped with first standby potentially on every query
+	// that could be considered an issue depending on requirements
+	project := bson.M{
+		"$project": bson.M{
+			"name":      1,
```

Comment:
We don't have to use aggregation if we just maintain another property called `votes` (integer) in the validator documents.

Reply:
added vote count as a part of adding extra collection